src/views/Register.vue

```diff
@@ -0,0 +1,131 @@
+                                    <label for="username">Username:</label>
+                                    <input id="username" v-model="username" type="text" />
+                                </div>
+                                <div class="user">
+                                    <label for="username">Email:</label>
+                                    <input id="username" v-model="email" type="text" />
+                                </div>
+                                <div class="pass">
+                                    <label for="password">Password:</label>
+                                    <input id="password" v-model="password" type="password" />
+                                </div>
+                                <div class="con-pass">
+                                    <label for="confirmPassword">Confirm Password:</label>
+                                    <input id="confirmPassword" v-model="confirmPassword" type="password" />
+                                </div>
+                                <div>
+                                    <button type="submit">Register</button>
+                                </div>
+                            </form>
+                        </div>
+                    </md-card-actions>
+                </md-card-media-actions>
+            </md-card>
+        </div>
+    </layout>
+</template>
+
+<script lang="ts">
+import firebase from "firebase";
+import DialogBox from "@/components/DialogBox/DialogBox";
+import { Vue, Component } from 'vue-property-decorator';
+import GarnBarnApi from "@/services/GarnBarnApi/GarnBarnApi";
+import GarnBarnApiConfig from "@/GarnBarnApiConfig.json";
+import axios from 'axios';
+import sha1 from 'sha1';
+
+@Component
+export default class Register extends Vue {
+    user: firebase.User | null = null;
+    username = '';
+    password = '';
+    confirmPassword = '';
+    email = ''
+
+    checkCompromisedPassword(): boolean {
+        const hashedPassword: string = sha1(this.password);
+        axios
+        .post(`${GarnBarnApiConfig.apiPrefix}/api/v1/account/compromised`, { "hashedPassword" : hashedPassword})
+        .then(response => {
+          return true
+        })
+        .catch(error => {
+          return false
+        });
+        return false
+    }
+
+    register(): void {
+        if (!this.checkCompromisedPassword()) {
+            alert('Your passoword is compromised')
+        }
+        firebase
+            .auth()
+            .createUserWithEmailAndPassword(this.email, this.password)
+            .then((res) => {
+                res.user!
+                    .updateProfile({
+                        displayName: this.username
+                    })
+                    .then(() => {
+                        this.$router.push('/signin')
+                    });
+            })
+            .catch((error) => {
+                alert(error.message);
+            });
```

Comment:
In case of `this.checkCompromisedPassword` return `false`

The user is still being created.